examples/user_guide/03-Customizing_Plots.ipynb

```diff
@@ -63,13 +63,87 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "%%output size=150\n",
-    "%%opts Curve  [height=100 width=600 xaxis=None tools=['hover']]\n",
+    "%%opts Curve [height=100 width=600 xaxis=None tools=['hover']]\n",
     "%%opts Curve (color='red' line_width=1.5)\n",
     "%%opts Spikes [height=100 width=600 yaxis=None] (color='grey' line_width=0.25)\n",
-    "curve  = hv.Curve( spike_train, 'milliseconds', vdims='Hertz')\n",
+    "curve = hv.Curve( spike_train, 'milliseconds', vdims='Hertz')\n",
+    "spikes = hv.Spikes(spike_train, 'milliseconds', vdims=[])\n",
+    "(curve + spikes).cols(1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here it is, again, with the `.options` method, which is the simplest way of customizing because it will automatically deduce whether an option is a style, plot, or norm option. In addition, `.options` is portable across notebook environments and `.py` scripts, and also will be saved on export, unlike `%%opts`."
```

Comment:
What does it mean to be "saved on export"?  To a .py script?  If so, maybe "is portable across notebook environments and `.py` scripts, including exporting the notebook to a `.py` file, unlike `%%opts`.".

Reply:
![image](https://user-images.githubusercontent.com/15331990/43751284-1e500156-99b1-11e8-80ad-48b31244c9ac.png)

I remember when I first started using HoloViews, this was one of the main issues I encountered, and I had little clue on how to use port the magic to .py script.

Reply:
Ah, here you need to move `hv.renderer` to a subsequent cell.  The `%%opts` cell magic is only applied to the *output* of the cell, when it's displayed, not to everything in it.

Reply:
Sorry I wasn't clear -- renderer.save() *should* keep *all* options that have been modified on the object; if it didn't, that would be a very strange bug.  But the point here is that you're doing renderer.save() on an object that has *not yet been modified by cell magic*, which is why you are failing to see any effect.  At that point, the cell magic application is still in the future! You have to do the renderer.save() *after* the cell has been displayed, because cell magics are applied only on the output of the cell, which hasn't yet occurred by the time you are saving it.  So all of the wordings you are suggesting are incorrect; %%opts *does* change the object in a way that will export; you just have to make sure that (a) the object is in fact affected by the cell magic (by returning it as the output of the cell), and (b) that you don't save it until *after* it has been returned as the output of the cell.  Does that make sense now?

Reply:
Okay, thanks for clarifying! I see your point now and learned something new! I wonder if I'm the only one who encountered this before? If not maybe should have it as another FAQ?

```
Q: Why isn't my %%opts magic applied when exported?
A: You're probably doing renderer.save() on an object that has not yet been modified by cell magic, which is why you are failing to see any effect. At that point, the cell magic application is still in the future! You have to do the renderer.save() after the cell has been displayed, because cell magics are applied only on the output of the cell, which hasn't yet occurred by the time you are saving it. So all of the wordings you are suggesting are incorrect; %%opts does change the object in a way that will export; you just have to make sure that (a) the object is in fact affected by the cell magic (by returning it as the output of the cell), and (b) that you don't save it until after it has been returned as the output of the cell. Does that make sense now?
```
![image](https://user-images.githubusercontent.com/15331990/43786901-a1528dd6-9a1e-11e8-9fc6-ecda3e47e205.png)

Reply:
Maybe make it more general:

Q: Why isn't my %%opts cell magic being applied to my HoloViews object?
A: %%opts is convenient because it tab-completes, but it can be confusing because of the "magic" way that it works.  Specifically, if you use it at the top of a Jupyter notebook cell, the indicated options will be applied to the *return value* of that cell, if it's a HoloViews object.  So, if you want a given object to get customized, you need to make sure it is returned from the cell, or the options won't *ever* be applied, and you should only access it *after* it has been returned, or the options won't *yet* have been applied.  For instance, if you use `renderer.save()` to export an object and only *then* return that object as the output of a cell, the exported object won't have the options applied, because they don't get applied until the object is returned (during IPython's "display hooks" processing).  So to make sure that options get applied, (a) return the object from a cell, and then (b) access it (e.g. for exporting) *after* the object has been returned.

Reply:
Also, another FAQ:

Q: Why are my `.options()` settings not having any effect?
A: By default, `.options()` returns a *copy* of your object, rather than modifying your original object. In HoloViews, making a copy of the object is cheap, because only the metadata is copied, not the data, and returning a copy makes it simple to work with a variety of differently customized versions of any given object.  You can pass `clone=False` to `.options()` if you wish to modify the object in place.